plugins/outputs/stackdriver/stackdriver.go

```diff
@@ -156,11 +159,19 @@ func (s *Stackdriver) Write(metrics []telegraf.Metric) error {
 
 			metricKind, err := getStackdriverMetricKind(m.Type())
 			if err != nil {
-				s.Log.Errorf("Get metric failed: %s", err)
+				s.Log.Errorf("Get kind for metric '%s' type '%T' field '%s' failed: %s", m.Name(), m.Type(), f, err)
 				continue
 			}
 
-			timeInterval, err := getStackdriverTimeInterval(metricKind, StartTime, m.Time().Unix())
+			startTime, endTime := getStackdriverIntervalEndpoints(
+				metricKind,
+				value,
+				m,
+				f,
+				s.counterCache,
+			)
```

Comment:
Can we maybe copy the content of `getStackdriverIntervalEndpoints` here as it saves a few lines and makes reading easier. Not a must, but I think it will save us some lines.

Reply:
actually my goal here was to make this easier to add a test for.  In penance... I'm adding a test. :)

Reply:
Ok. Then please do not split the params into lines and keep everything on a single line for readability. I guess that will fit in 120 cols... ;-)

Reply:
👍